reana_client/utils.py

```diff
@@ -173,6 +180,49 @@ def _prepare_kwargs(reana_yaml):
         raise e
 
 
+def _validate_environment_tags(reana_yaml):
+    """Validate environments in REANA specification file according to workflow type.
+
+    :param reana_yaml: Dictionary which represents REANA specifications file.
+    :return: True if environment tags are valid.
+    """
+    workflow_type = reana_yaml["workflow"]["type"]
+
+    if workflow_type == "serial":
+        workflow_steps = reana_yaml["workflow"]["specification"]["steps"]
+        _validate_serial_workflow_environment(workflow_steps)
+    elif workflow_type == "yadage":
+        logging.info("Not Implemented yet.")
+    elif workflow_type == "cwl":
+        logging.info("Not Implemented yet.")
+
+    return True
+
+
+def _validate_serial_workflow_environment(workflow_steps):
+    """Validate environments in REANA serial workflow.
+
+    :param workflow_steps: List of dictionary which represents different steps involved in workflow.
+    :raises Warning: Warns user if valid tags are not used in serial workflow steps.
+    """
+    for step in workflow_steps:
+        environment = step["environment"].split(":")
+        environment_tag = environment[-1]
+        if len(environment) == 1:
+            logging.warning(
+                "Include a tag in the enviornment {}".format(environment[0])
+            )
+            sys.exit(1)
+        elif len(environment) == 2 and environment_tag in WARNING_TAGS:
+            logging.warning(
+                "Using {} in {} is not recommended.".format(
+                    environment_tag, environment[0]
+                )
+            )
+            sys.exit(1)
+    return True
```

Comment:
This value is not being used, should we return a boolean? If so, maybe we shouldn't stop the execution (`sys.exit(1)`) and return `False` when the validation fails? cc/ @tiborsimko

Reply:
Boolean is not required, I can omit return. 👍

Reply:
We shall probably refactor the exit-or-not strategy soon, after we shall introduce those additional tests for environment images.